ironfish/src/network/peers/peerManager.ts

```diff
@@ -535,10 +535,11 @@ export class PeerManager {
 
   /**
    * Returns true if the total number of connected peers is less
-   * than the target amount of peers
+   * than the target and max amount of peers
    */
   canCreateNewConnections(): boolean {
-    return this.getPeersWithConnection().length < this.targetPeers
+    const currentPeers = this.getPeersWithConnection().length
+    return currentPeers < this.targetPeers && currentPeers < this.maxPeers
```

Comment:
Should we throw, or clamp? What's the worst case scenario if it happens to clamp it without your knowledge? I think it would probably be fine right?

Reply:
@wd021 I think clamping the value in the PeerNetwork constructor instead of throwing an error would be fine. The targetPeers and maxPeers values aren't currently set up to change dynamically (in other words, if you run config:set while the node is running, they won't update in PeerNetwork), so I don't think we need to change the setConfig or getConfig routes.

Reply:
updated